tests/integration/otel_thrift_tests.py

```diff
@@ -0,0 +1,963 @@
+            {"server_timeout.default": "1 hour"}
+        )
+        with serve_thrift(handler, TestService, baseplate_observer=timeout_observer) as server:
+            with baseplate_thrift_client(
+                server.endpoint, TestService, timeout="1000 seconds"
+            ) as context:
+                with context.example_service.retrying(
+                    attempts=3, budget=retry_timeout_seconds
+                ) as svc:
+                    svc.example()
+
+        self.assertAlmostEqual(handler.context.deadline_budget, retry_timeout_seconds)
+
+
+class ThriftHealthcheck(GeventPatchedTestCase):
+    def test_v2_client_v1_server(self):
+        class Handler(BaseplateService.Iface):
+            def is_healthy(self, context):
+                return True
+
+        handler = Handler()
+
+        with serve_thrift(handler, BaseplateService) as server:
+            with baseplate_thrift_client(server.endpoint, BaseplateServiceV2) as context:
+                healthy = context.example_service.is_healthy(
+                    request=IsHealthyRequest(probe=IsHealthyProbe.READINESS),
+                )
+                self.assertTrue(healthy)
+
+    def test_v2_client_v2_server(self):
+        class Handler(BaseplateServiceV2.Iface):
+            def __init__(self):
+                self.probe = None
+
+            def is_healthy(self, context, req=None):
+                self.probe = req.probe
+                return True
+
+        handler = Handler()
+
+        with serve_thrift(handler, BaseplateServiceV2) as server:
+            with baseplate_thrift_client(server.endpoint, BaseplateServiceV2) as context:
+                healthy = context.example_service.is_healthy(
+                    request=IsHealthyRequest(probe=IsHealthyProbe.LIVENESS),
+                )
+                self.assertTrue(healthy)
+                self.assertEqual(handler.probe, IsHealthyProbe.LIVENESS)
+
+
+class ThriftErrorReplacementTests(GeventPatchedTestCase):
+    def test_server_replaces_unhandled_errors(self):
+        """The server span should start/stop appropriately."""
+
+        class Handler(TestService.Iface):
+            def example(self, context):
+                raise Exception("foo")
+
+        handler = Handler()
+
+        with serve_thrift(handler, TestService) as server:
+            with raw_thrift_client(server.endpoint, TestService) as client:
+                with self.assertRaises(Error) as exc_info:
+                    client.example()
+        self.assertEqual(exc_info.exception.code, ErrorCode.INTERNAL_SERVER_ERROR)
+
+
+class ThriftPrometheusMetricsTests(GeventPatchedTestCase):
+    def reset_metrics(self, metrics):
+        if not metrics:
+            return
+
+        try:
+            metrics.get_active_requests_metric().clear()
+            metrics.get_latency_seconds_metric().clear()
+            metrics.get_requests_total_metric().clear()
+        except Exception:
+            pass
```

Comment:
How does this exception happen? Is it possible to catch a specific error?

Reply:
I'll look into the exceptions thrown here and clean this up.

Reply:
This method wasn't called so I just removed it.

I also removed it from the `thrift_tests.py` file as well as that's where we lifted most of these tests.